packages/eas-cli/src/run/ios/IosRunner.ts

```diff
@@ -0,0 +1,264 @@
+        if (device.isAvailable) {
+          iosSimulators.push({
+            ...device,
+            runtime,
+            osVersion,
+            windowName: `${device.name} (${osVersion})`,
+            osType: 'iOS' as const,
+            state: device.state as 'Booted' | 'Shutdown',
+            lastBootedAt: device.lastBootedAt ? new Date(device.lastBootedAt) : undefined,
+          });
+        }
+      }
+    }
+  }
+  return iosSimulators;
+}
+
+function parseSimControlJsonResults(input: string): any {
+  try {
+    return JSON.parse(input);
+  } catch (error: any) {
+    // Nov 15, 2020: Observed this can happen when opening the simulator and the simulator prompts the user to update the xcode command line tools.
+    // Unexpected token I in JSON at position 0
+    if (error.message.includes('Unexpected token')) {
+      Log.error(`Apple's simctl returned malformed JSON:\n${input}`);
+    }
+    throw error;
+  }
+}
+
+async function ensureSimulatorBootedAsync(simulator: IosSimulator): Promise<void> {
+  if (simulator.state === 'Booted') {
+    return;
+  }
+
+  await simctlAsync(['boot', simulator.udid]);
+}
+
+async function openSimulatorAppAsync(simulatorUdid: string): Promise<void> {
+  const args = ['-a', 'Simulator'];
+  if (simulatorUdid) {
+    // This has no effect if the app is already running.
+    args.push('--args', '-CurrentDeviceUDID', simulatorUdid);
+  }
+  await spawnAsync('open', args);
+}
+
+async function launchAppOnIosSimulatorAsync(
+  simulatorUdid: string,
+  bundleIdentifier: string
+): Promise<void> {
+  Log.newLine();
+  Log.log('Launching your app...');
+
+  await simctlAsync(['launch', simulatorUdid, bundleIdentifier]);
+
+  Log.succeed('Successfully launched your app!');
+}
+
+async function getAppBundleIdentifierAsync(appPath: string): Promise<string> {
+  const { stdout } = await spawnAsync('xcrun', [
+    'plutil',
+    '-extract',
+    'CFBundleIdentifier',
+    'raw',
+    path.join(appPath, 'Info.plist'),
+  ]);
+
+  if (!stdout) {
+    throw new Error(
+      `Could not read app bundle identifier from ${path.join(appPath, 'Info.plist')}`
+    );
+  }
+
+  return stdout.trim();
+}
+
+// I think the app can be open while no simulators are booted.
+async function waitForSimulatorAppToStartAsync(
+  maxWaitTimeMs: number,
+  intervalMs: number
+): Promise<void> {
+  Log.newLine();
+  Log.log('Waiting for Simulator app to start...');
+
+  const startTime = Date.now();
+  while (Date.now() - startTime < maxWaitTimeMs) {
+    if (await isSimulatorAppRunningAsync()) {
+      return;
+    }
+    await sleepAsync(Math.min(intervalMs, Math.max(maxWaitTimeMs - (Date.now() - startTime), 0)));
+  }
+  throw new Error('Timed out waiting for simulator app to start.');
+}
+
+async function isSimulatorAppRunningAsync(): Promise<boolean> {
+  const { stdout } = await spawnAsync('pgrep', ['Simulator']);
+
+  if (stdout.split('\n').length < 4) {
+    return false;
+  }
+
+  return true;
+}
```

Comment:
This is really ugly I need to find a better way to do it. I will probably use `@expo/osascript` as Expo CLI does